src/Http/Http.Extensions/src/EndpointParameterMetadataContext.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Http.Metadata;
+
+/// <summary>
+/// Represents the information accessible during endpoint creation by types that implement <see cref="IEndpointParameterMetadataProvider"/>.
+/// </summary>
+public class EndpointParameterMetadataContext
+{
+    /// <summary>
+    /// Gets the parameter of the route handler delegate of the endpoint being created.
+    /// </summary>
+    public ParameterInfo? Parameter { get; set; } // set to allow re-use
```

Comment:
I think changing this to `set;` is unnecessary. Compared to the other costs involved in calling `PopulateMetadataForParameter` like calling `MakeGenericMethod`, allocating the context is inconsequential. I prefer keeping the API more consistent.